lib/puppet/provider/foreman_auth/rest_v3.rb

```diff
@@ -0,0 +1,274 @@
+Puppet::Type.type(:foreman_auth).provide(
+  :rest_v3,
+  parent: Puppet::Type.type(:foreman_resource).provider(:rest_v3)
+) do
+  confine :feature => [:json, :oauth]
+
+  mk_resource_methods
+
+  def b2i(value)
+    value ? '1' : '0'
+  end
+
+  def source
+    @source ||= begin
+      r = request(:get,
+                  'api/v2/auth_source_ldaps',
+                  search: %(name="#{resource[:name]}"))
+      unless success?(r)
+        raise Puppet::Error,
+              "LDAP #{resource[:name]} cannot be retrieved: #{error_message(r)}"
+      end
+      JSON.parse(r.body)['results'][0]
+    end
+  end
+
+  def id
+    source ? source['id'] : nil
+  end
+
+  def create
+    post_data = {
+      auth_source_ldap: {
+        account:           resource[:account],
```

Comment:
You could put any defaults here. See https://github.com/voxpupuli/puppet-proxysql/blob/00eb03323c52fe0f2665fdaf3e60937eb7e22867/lib/puppet/provider/proxy_mysql_server/proxysql.rb#L59 for example